auth/sasl/authenticatee_test.go

```diff
@@ -0,0 +1,98 @@
+package sasl
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mesos/mesos-go/auth/callback"
+	"github.com/mesos/mesos-go/auth/sasl/mech/crammd5"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesos/mesos-go/messenger"
+	"github.com/mesos/mesos-go/upid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"golang.org/x/net/context"
+)
+
+type MockTransport struct {
+	*messenger.MockedMessenger
+}
+
+func (m *MockTransport) Send(ctx context.Context, upid *upid.UPID, msg proto.Message) error {
+	return m.Called(mock.Anything, upid, msg).Error(0)
+}
+
+func TestAuthticatee_validLogin(t *testing.T) {
+	assert := assert.New(t)
+	ctx := context.TODO()
+	client := upid.UPID{
+		ID:   "someFramework",
+		Host: "b.net",
+		Port: "789",
+	}
+	server := upid.UPID{
+		ID:   "serv",
+		Host: "a.com",
+		Port: "123",
+	}
+	tpid := upid.UPID{
+		ID:   "sasl_transport",
+		Host: "g.org",
+		Port: "456",
+	}
+	handler := callback.HandlerFunc(func(cb ...callback.Interface) error {
+		for _, c := range cb {
+			switch c := c.(type) {
+			case *callback.Name:
+				c.Set("foo")
+			case *callback.Password:
+				c.Set([]byte("bar"))
+			case *callback.Interprocess:
+				c.Set(server, client)
+			default:
+				return &callback.Unsupported{c}
+			}
+		}
+		return nil
+	})
+	var transport *MockTransport
+	factory := transportFactoryFunc(func() messenger.Messenger {
```

Comment:
I'm not sure I understand your suggestion. I used a factory function because I want a new instance of the messenger implementation for each new authentication attempt. If I instantiated the messenger once and just kept using that then, yes, I could get rid of the factory function. This is modeled on the design used in the Mesos native authenticatee, which spawns a new process (get's a new authentication PID) for each.

What's somewhat interesting to think about is to have the authentication messenger share the same HTTP transport as the driver messenger. However, I wouldn't want the authentication messenger to Start() or Stop() the HTTP transport since the driver expects to control the lifecycle of the transport that it created. But that way there'd be only a single HTTP listener handling traffic between the framework and the scheduler. It could probably be done fairly simply creating a SharedTransport implementation that wraps an existing transport, turns Start() into a noop, and re-implements Stop() to Uninstall the handlers registered via the wrapper (but that's it, it wouldn't actually stop the wrapped transport).